src/core/qgsattributetableconfig.cpp

```diff
@@ -242,6 +248,12 @@ Qt::SortOrder QgsAttributeTableConfig::sortOrder() const
 
 void QgsAttributeTableConfig::setSortOrder( Qt::SortOrder sortOrder )
 {
+  // fix https://hub.qgis.org/issues/15803
+  if ( sortOrder != Qt::AscendingOrder && sortOrder != Qt::DescendingOrder )
```

Comment:
I would log this anomalies... but I didn't because seems there is a general behavior to leave container classes silent. IMHO reading a wrong value from the project would be notified, at list in the log. What do you think @m-kuhn ?